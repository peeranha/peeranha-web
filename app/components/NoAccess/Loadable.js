/**
 *
 * Asynchronously loads the component for NoAccess
 *
 */

import Loadable from 'react-loadable';

export default Loadable({
  loader: () => import('./index'),
  loading: () => null,
});
