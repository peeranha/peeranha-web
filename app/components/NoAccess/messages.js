/*
 * NoAccess Messages
 *
 * This contains all the text for the NoAccess component.
 */

import { defineMessages } from 'react-intl';

export default defineMessages({
  errorMessage: {
    id: 'app.components.NoAccess.errorMessage',
    defaultMessage: 'No access to complete this action!',
  },
});
