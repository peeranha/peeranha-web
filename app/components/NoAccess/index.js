/**
 *
 * NoAccess
 *
 */

import React from 'react';

import { FormattedMessage } from 'react-intl';
import messages from './messages';

function NoAccess() {
  return (
    <div className="container">
      <div className="text-center pt-3">
        <FormattedMessage {...messages.errorMessage} />
      </div>
    </div>
  );
}

export default NoAccess;
