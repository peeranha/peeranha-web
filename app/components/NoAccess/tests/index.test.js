import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { IntlProvider } from 'react-intl';

import NoAccess from '../index';

describe('<NoAccess />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer
      .create(
        <IntlProvider locale="en" key="en" messages={{}}>
          <NoAccess />
        </IntlProvider>,
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});
