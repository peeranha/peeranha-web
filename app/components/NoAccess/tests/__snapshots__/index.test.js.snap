// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoAccess /> should match the snapshot 1`] = `
<div
  className="container"
>
  <div
    className="text-center pt-3"
  >
    <span>
      No access to complete this action!
    </span>
  </div>
</div>
`;
